build.sbt

```diff
@@ -24,32 +28,35 @@ resolvers := Seq("Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/r
   Seq("Typesafe Repo" at "http://repo.typesafe.com/typesafe/releases/")
 
 libraryDependencies ++= Seq(
-  "com.codacy" %% "codacy-engine-scala-seed" % "3.0.9",
-  "com.vladsch.flexmark" % "flexmark-all" % "0.34.8",
-  "org.specs2" %% "specs2-core" % "4.2.0" % Test)
+  "com.codacy" %% "codacy-engine-scala-seed" % "3.0.296",
+  "com.vladsch.flexmark" % "flexmark-all" % "0.50.20",
+  "org.specs2" %% "specs2-core" % "4.6.0" % Test)
 
-scalacOptions ++= Common.compilerFlags
 scalacOptions.in(Compile, console) --= Seq("-Ywarn-unused", "-Ywarn-unused:imports", "-Xfatal-warnings")
 scalacOptions in Test ++= Seq("-Yrangepos")
 enablePlugins(JavaAppPackaging)
 
 enablePlugins(DockerPlugin)
 
-version in Docker := "1.0.0-SNAPSHOT"
-
-organization := "com.codacy"
-
-lazy val toolVersion = taskKey[String]("Retrieve the version of the underlying tool from patterns.json")
+lazy val toolVersion = settingKey[String]("The version of the underlying tool retrieved from patterns.json")
 toolVersion := {
-  import better.files.File
-  import play.api.libs.json.{JsString, JsValue, Json}
-
-  val jsonFile = resourceDirectory.in(Compile).value / "docs" / "patterns.json"
-  val patternsJsonValues = Json.parse(File(jsonFile.toPath).contentAsString).as[Map[String, JsValue]]
+  case class Patterns(name: String, version: String)
+  implicit val patternsIso: IsoLList[Patterns] =
+    LList.isoCurried((p: Patterns) => ("name", p.name) :*: ("version", p.version) :*: LNil) {
+      case (_, n) :*: (_, v) :*: LNil => Patterns(n, v)
+    }
+
+  val jsonFile = (resourceDirectory in Compile).value / "docs" / "patterns.json"
+  val json = Parser.parseFromFile(jsonFile)
+  val patterns = json.flatMap(Converter.fromJson[Patterns])
+  patterns.map(_.version).getOrElse(throw new Exception("Failed to retrieve version from docs/patterns.json"))
+}
 
-  patternsJsonValues.collectFirst {
-    case ("version", JsString(ver)) => ver
-  }.getOrElse(throw new Exception("Failed to retrieve version from docs/patterns.json"))
+lazy val writeStylelintConf = taskKey[Unit]("Create .stylelint-version file from toolVersion")
```

Comment:
this `taskKey` output a file, we should not rely on the order of execution but on the output of the tasks.
e.g.:
```
taskKey[File]
```
and later on:
```
writeStylelintConf.value
```
avoiding to repeat the filename as well

Reply:
I avoided repeating the filename many times. Couldn't make `writeStylelintConf` a `taskKey[File]` because I need to call `.value` inside a setting definition and there is no easy way to do it.
I'm maintaining current implementation to not do hacky things (like custom implementation in a .scala helper)

Reply:
Not 100% sure, but from a first look I think you can avoid to use the `mappings` and use directly `resourceManaged`(for generated files) or `unmanagedResource`(for adding a directory to the resource path) that is more semantically correct and will allow you to feed with a taskKey IIRC.

Reply:
I managed to solve it maintaining `mappings` instead of moving to `resourceManaged` and `resourceManaged`. 
I made `stylelintVersionFile` a `Setting` instead of a `Task`.
and I'm using `zip` to get its value:
```scala
resourceDirectory
  .in(Compile)
  .zip(stylelintVersionFile)
```

Reply:
:+1: